plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/NioHttpServerTransport.java

```diff
@@ -152,6 +152,11 @@ protected HttpServerChannel bind(InetSocketAddress socketAddress) throws IOExcep
         return nioGroup.bindServerChannel(socketAddress, channelFactory);
     }
 
+    @Override
+    protected void onException(HttpChannel channel, Exception cause) {
+        super.onException(channel, cause);
```

Comment:
hmm this is obsolete?

Reply:
This is needed so that the `verify` call [here](https://github.com/elastic/elasticsearch/pull/41466/files#diff-95e031a2bceadddcd5b6200b23efd8a6R354) works. But I guess we could add a comment pointing that out :)?

Reply:
It's needed so that the `HttpReadWriteHandler` can call `onException`. But in the update, I will just make the method public similar to the `onException` method in `TcpTransport`.